lightning/src/ln/peer_handler.rs

```diff
@@ -1668,6 +1688,30 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 
 			for (descriptor, peer_mutex) in peers.iter() {
 				self.do_attempt_write_data(&mut (*descriptor).clone(), &mut *peer_mutex.lock().unwrap());
+
+				// Only see if we have room for onion messages after we've written all channel messages, to
+				// ensure the latter take priority.
+				let (peer_node_id, mut om_buffer_slots_avail) = {
+					let peer = peer_mutex.lock().unwrap();
+					if let Some(peer_node_id) = peer.their_node_id {
+						(peer_node_id.clone(), peer.onion_message_buffer_slots_available())
+					} else { continue; }
+				};
```

Comment:
Could this be folded into the `loop` so it doesn't need to be repeated at the end of the `loop`?

Also, looks like we're locking the peer mutex a few times throughout this loop, so may be worth just holding lock throughout the entire iteration?
- in `get_peer_for_forwarding!` for each message,
- for `do_attempt_write_data`, and
- when updating `om_buffer_slots_avail`

Reply:
> Also, looks like we're locking the peer mutex a few times throughout this loop, so may be worth just holding lock throughout the entire iteration?
>  *  in get_peer_for_forwarding! for each message,

I don't think we can do this for `get_peer_for_forwarding` because it takes the `peer` lock internally. 

I think your first piece of feedback did get rid of a lock or two